### ISSUE TITLE
Errors when num_workers is set to be value bigger than 0 in torch.utils.data.DataLoader

### ISSUE THREAD

**Issue by eriche2016** [pytorch/pytorch#1897]
I have a dataset whose labels are from 0 to 9. And i wrap it using torch.utils.data.DataLoader, if i set num_workers to be 0, everything works fine, However if it is set to be 2, then the labels batch (a 1-D ***byte tensor***)it loads at some epoch always seems to be bigger than 9, which seems to be 255. what causes this problem? any help? ( p.s., my dataset is .h5 file).

**Comment by eriche2016** [comment]
problem solved.
